mongo_connector/connector.py

```diff
@@ -330,7 +330,8 @@ def run(self):
         else:       # sharded cluster
             while self.can_run is True:
 
-                for shard_doc in main_conn['config']['shards'].find():
+                for shard_doc in retry_until_ok(
+                        lambda: list(main_conn.config.shards.find())):
```

Comment:
Good point. If there are so many shards that we have to ask for a second batch, that will be a problem.

However, this might be a better approach: https://docs.mongodb.com/master/reference/command/listShards/

Annoyingly, that page doesn't provide any example output for the command. Though, running the command locally produces output like this:

```json
{
        "shards" : [
                {
                        "_id" : "4a0f71de-e0e1-42da-86ee-4a70ebdeae8a",
                        "host" : "localhost:1028",
                        "state" : 1
                },
                {
                        "_id" : "7c3da9c7-5a49-4539-bc9b-3343e8984ecf",
                        "host" : "localhost:1029",
                        "state" : 1
                },
                {
                        "_id" : "7fce9155-6c41-4cc3-9c54-477330d49c56",
                        "host" : "localhost:1030",
                        "state" : 1
                }
        ],
        "ok" : 1
}
```